app/controllers/bulk_downloads_controller.rb

```diff
@@ -23,6 +24,14 @@ def create
     # Convert sample ids to pipeline run ids.
     begin
       sample_ids = create_params[:sample_ids]
+
+      # Max samples check.
+      max_samples_allowed = get_app_config(AppConfig::MAX_SAMPLES_BULK_DOWNLOAD)
+
+      if sample_ids.length > Integer(max_samples_allowed)
```

Comment:
I think this is in danger of TypeError for `Integer(nil)` as well?

Reply:
Same here.